src/main/java/com/fulcrumgenomics/util/DemuxFastqs.java

```diff
@@ -677,14 +697,24 @@ void convertQuality(final byte[] quals, final FastqQualityFormat version) {
             return sampleBarcodeQualities;
         }
 
-        private String getMolecularBarcode(final List<FastqRecord>  records) {
-            final byte[][] molecularBarcode = new byte[records.size()][];
+        private byte[][] getMolecularBarcodes(final List<FastqRecord> records) {
+            final byte[][] molecularBarcodes = new byte[records.size()][];
```

Comment:
Not related to this change, but why not just use a `List<byte[]>` here so you don't have to futz with copying it below?